sklearn/naive_bayes.py

```diff
@@ -53,8 +53,8 @@ def _joint_log_likelihood(self, X):
         I.e. ``log P(c) + log P(x|c)`` for all rows x of X, as an array-like of
         shape (n_samples, n_classes).
 
-        Input is passed to _joint_log_likelihood as-is by predict,
-        predict_proba and predict_log_proba.
+        Input is handed over to _joint_log_likelihood by predict, predict_proba
+        and predict_log_proba after being passed through _check_X.
```

Comment:
I think we can change the order a little to clarify this statement:

```suggestion
        predict, predict_proba, and predict_log_proba checks the input and passes
        it along to _joint_log_likelihood.
```

Reply:
"Checks the input" may mean `_check_X(X)`. What I mean by "passed through" is `X = _check_X(X)`. This difference is crucial, because `BernoulliNB` performs binarisation inside `_check_X`. This was not obvious for me when I was writing a meta-estimator: I couldn't just call `BernoulliNB._joint_log_likelihood`, I had to pull X through `BernoulliNB._check_X`.

Reply:
What about:

> predict, predict_proba, and predict_log_proba passes the input through `_check_X` and handles it over to _joint_log_likelihood

Reply:
Done. Please review.